src/plugins/intel_cpu/thirdparty/mlas/CMakeLists.txt

```diff
@@ -0,0 +1,251 @@
+     "${MLAS_SRC_DIR}/intrinsics/avx/*.cpp"
+   )
+   set_source_files_properties(${mlas_platform_srcs_avx} PROPERTIES COMPILE_FLAGS "/arch:AVX")
+
+   file(GLOB_RECURSE mlas_platform_srcs_avx2 CONFIGURE_DEPENDS
+     "${MLAS_SRC_DIR}/intrinsics/avx2/*.cpp"
+   )
+   set_source_files_properties(${mlas_platform_srcs_avx2} PROPERTIES COMPILE_FLAGS "/arch:AVX2")
+   target_sources(mlas PRIVATE
+      ${MLAS_SRC_DIR}/dgemm.cpp
+      ${mlas_platform_srcs_avx}
+      ${mlas_platform_srcs_avx2}
+      ${MLAS_SRC_DIR}/qgemm_kernel_avx2.cpp
+      ${MLAS_SRC_DIR}/qgemm_kernel_sse.cpp
+      ${MLAS_SRC_DIR}/qgemm_kernel_sse41.cpp
+      ${MLAS_SRC_DIR}/intrinsics/avx512/quantize_avx512f.cpp
+      ${MLAS_SRC_DIR}/amd64/QgemmU8S8KernelAvx2.asm
+      ${MLAS_SRC_DIR}/amd64/QgemmU8U8KernelAvx2.asm
+      ${MLAS_SRC_DIR}/amd64/QgemmU8X8KernelAvx2.asm
+      ${MLAS_SRC_DIR}/amd64/QgemmU8X8KernelAvx512Core.asm
+      ${MLAS_SRC_DIR}/amd64/QgemvU8S8KernelAvx2.asm
+      ${MLAS_SRC_DIR}/amd64/QgemvU8S8KernelAvx512Core.asm
+      ${MLAS_SRC_DIR}/amd64/QgemvU8S8KernelAvx512Vnni.asm
+      ${MLAS_SRC_DIR}/amd64/QgemvU8S8KernelAvxVnni.asm
+      ${MLAS_SRC_DIR}/amd64/ConvSymKernelAvx2.asm
+      ${MLAS_SRC_DIR}/amd64/ConvSymKernelAvx512Core.asm
+      ${MLAS_SRC_DIR}/amd64/DgemmKernelSse2.asm
+      ${MLAS_SRC_DIR}/amd64/DgemmKernelAvx.asm
+      ${MLAS_SRC_DIR}/amd64/DgemmKernelFma3.asm
+      ${MLAS_SRC_DIR}/amd64/DgemmKernelAvx512F.asm
+      ${MLAS_SRC_DIR}/amd64/SgemmKernelSse2.asm
+      ${MLAS_SRC_DIR}/amd64/SgemmKernelAvx.asm
+      ${MLAS_SRC_DIR}/amd64/SgemmKernelM1Avx.asm
+      ${MLAS_SRC_DIR}/amd64/SgemmKernelFma3.asm
+      ${MLAS_SRC_DIR}/amd64/SgemmKernelAvx512F.asm
+      ${MLAS_SRC_DIR}/amd64/SconvKernelSse2.asm
+      ${MLAS_SRC_DIR}/amd64/SconvKernelAvx.asm
+      ${MLAS_SRC_DIR}/amd64/SconvKernelFma3.asm
+      ${MLAS_SRC_DIR}/amd64/SconvKernelAvx512F.asm
+      ${MLAS_SRC_DIR}/amd64/SpoolKernelSse2.asm
+      ${MLAS_SRC_DIR}/amd64/SpoolKernelAvx.asm
+      ${MLAS_SRC_DIR}/amd64/SpoolKernelAvx512F.asm
+      ${MLAS_SRC_DIR}/amd64/sgemma.asm
+      ${MLAS_SRC_DIR}/amd64/cvtfp16a.asm
+      ${MLAS_SRC_DIR}/amd64/SoftmaxKernelAvx.asm
+      ${MLAS_SRC_DIR}/amd64/TransKernelFma3.asm
+      ${MLAS_SRC_DIR}/amd64/TransKernelAvx512F.asm
+      ${MLAS_SRC_DIR}/amd64/LogisticKernelFma3.asm
+      ${MLAS_SRC_DIR}/amd64/TanhKernelFma3.asm
+      ${MLAS_SRC_DIR}/amd64/ErfKernelFma3.asm
+    )
+  else()
+    target_sources(mlas PRIVATE
+      ${MLAS_SRC_DIR}/qgemm_kernel_sse.cpp
+      ${MLAS_SRC_DIR}/qgemm_kernel_sse41.cpp
+      ${MLAS_SRC_DIR}/i386/SgemmKernelSse2.asm
+      ${MLAS_SRC_DIR}/i386/SgemmKernelAvx.asm
+    )
+  endif()
+endfunction()
+
+if(MSVC)
+    set(onnxruntime_target_platform "x64")
+    setup_mlas_source_for_windows()
+else()
+    if(CMAKE_SYSTEM_PROCESSOR MATCHES "^(i.86|x86?)$")
+        set(X86 TRUE)
+    elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^(x86_64|amd64)$")
+        set(X86_64 TRUE)
+    endif()
```

Comment:
I see that you have also added ARM sources, while cmake script does not have it.